adapters/adview/adview.go

```diff
@@ -0,0 +1,167 @@
+package adview
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adViewAdapter struct {
+	EndpointTemplate template.Template
+}
+
+// Builder builds a new instance of the Adf adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+
+	urlTemplate, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adViewAdapter{
+		EndpointTemplate: *urlTemplate,
+	}
+	return bidder, nil
+}
+
+func (adapter *adViewAdapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	var validImps = make([]openrtb2.Imp, 0, len(request.Imp))
+	requestData := make([]*adapters.RequestData, 0, len(request.Imp))
+
+	for _, imp := range request.Imp {
+
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: err.Error(),
+			})
+			continue
+		}
+		//采用 adview
+		var advImpExt openrtb_ext.ExtImpAdview
+		if err := json.Unmarshal(bidderExt.Bidder, &advImpExt); err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: err.Error(),
+			})
+			continue
+		}
+
+		imp.TagID = advImpExt.MasterTagID //tagid means posid
+
+		//for adview bid request
+		if imp.Banner != nil {
+			if len(imp.Banner.Format) != 0 {
+				imp.Banner.H = &imp.Banner.Format[0].H
```

Comment:
Please add a json test for the `len(imp.Banner.Format) != 0` condition.

Reply:
ok, i added the banner-format.json test case.